src/components/secure-backup/generate-backup/index.tsx

```diff
@@ -0,0 +1,67 @@
+import * as React from 'react';
+
+import { bemClassName } from '../../../lib/bem';
+
+import { Button } from '@zero-tech/zui/components';
+import { IconArrowRight } from '@zero-tech/zui/icons';
+
+import '../styles.scss';
+
+const cn = bemClassName('secure-backup');
+
+export interface Properties {
+  recoveryKey: string;
+  hasCopied: boolean;
+  errorMessage: string;
+
+  onCopy: () => void;
+  onSave: () => void;
+}
+
+export class GenerateBackup extends React.Component<Properties> {
+  onSave = () => {
+    this.props.onSave();
+  };
+
+  onCopy = () => {
+    this.props.onCopy();
+  };
+
+  get copyButtonText() {
+    return this.props.hasCopied ? 'Copied' : 'Copy';
+  }
+
+  render() {
+    return (
+      <>
+        <div {...cn('recovery-key-container')}>
+          <p {...cn('primary-text')}>
+            This backup phrase can be entered to view past messages on <em>any device, anywhere</em>. Store it securely
+            and never share it with anyone:
+          </p>
+
+          {this.props.recoveryKey && <div {...cn('recovery-key')}>{this.props.recoveryKey}</div>}
```

Comment:
Do we ever render this when there's no recovery key? I.E., do we need these checks?

Reply:
Error message is needed but no the others aren't here. 

We should probably handle errors when generating the key
```
export function* generateBackup() {
  yield put(setBackupStage(BackupStage.GenerateBackup));
  const chatClient = yield call(chat.get);
  const key = yield call([chatClient, chatClient.generateSecureBackup]);
  yield put(setGeneratedRecoveryKey(key));
}
```